test/module/irohad/torii/torii_transport_command_test.cpp

```diff
@@ -0,0 +1,357 @@
+}
+
+/**
+ * @given torii service and command_service with empty status stream
+ * @when calling StatusStream on transport
+ * @then Ok status is eventually returned without any fault
+ *       and nothing is written to the status stream
+ */
+TEST_F(CommandServiceTransportGrpcTest, StatusStreamEmpty) {
+  grpc::ServerContext context;
+  iroha::protocol::TxStatusRequest request;
+
+  EXPECT_CALL(*command_service, getStatusStream(_))
+      .WillOnce(Return(rxcpp::observable<>::empty<std::shared_ptr<
+                           shared_model::interface::TransactionResponse>>()));
+
+  ASSERT_TRUE(transport_grpc->StatusStream(&context, &request, nullptr).ok());
+}
+
+/**
+ * @given torii service with changed timeout, a transaction
+ *        and a status stream with one NotRecieved status
+ * @when calling StatusStream
+ * @then ServerWriter calls Write method
+ *
+ */
+TEST_F(CommandServiceTransportGrpcTest, StatusStreamOnNotReceived) {
+  const std::chrono::milliseconds kInitialTimeout = 1ms;
+  const std::chrono::milliseconds kNonFinalTimeout = 1ms;
+  transport_grpc =
+      std::make_shared<torii::CommandServiceTransportGrpc>(command_service,
+                                                           status_bus,
+                                                           kInitialTimeout,
+                                                           kNonFinalTimeout,
+                                                           status_factory,
+                                                           transaction_factory,
+                                                           batch_parser,
+                                                           batch_factory);
+  grpc::ServerContext context;
+  iroha::protocol::TxStatusRequest request;
+  iroha::MockServerWriter<iroha::protocol::ToriiResponse> response_writer;
+  std::vector<std::shared_ptr<shared_model::interface::TransactionResponse>>
+      responses;
+  shared_model::crypto::Hash hash("1");
+  responses.emplace_back(status_factory->makeNotReceived(hash, {}));
+  EXPECT_CALL(*command_service, getStatusStream(_))
+      .WillOnce(Return(rxcpp::observable<>::iterate(responses)));
+  EXPECT_CALL(response_writer,
+              Write(Property(&iroha::protocol::ToriiResponse::tx_hash,
+                             StrEq(shared_model::crypto::toBinaryString(hash))),
+                    _))
+      .WillOnce(Return(true));
+  ASSERT_TRUE(transport_grpc
+                  ->StatusStream(
+                      &context,
+                      &request,
+                      reinterpret_cast<
+                          grpc::ServerWriter<iroha::protocol::ToriiResponse> *>(
+                          &response_writer))
+                  .ok());
+}
+
+/**
+ * @given torii service and number of invalid transactions
+ * @when calling StatusStream
+ * @then ServerWriter call Write method the same number of times
+ */
+TEST_F(CommandServiceTransportGrpcTest, StatusStream) {
```

Comment:
This test can be removed since it covers the same logic as `StatusStreamOnNotReceived` at the moment.